plugins/toolbox/src/components/Misc/Diff.tsx

```diff
@@ -0,0 +1,171 @@
+import { DiffEditor } from '@monaco-editor/react';
+import React, { useEffect, useState } from 'react';
+import { Button, FormControl, Grid } from '@material-ui/core';
+import { useStyles } from '../../utils/hooks';
+import * as monaco from 'monaco-editor';
+import { useEffectOnce } from 'react-use';
+import { Select, SelectItem } from '@backstage/core-components';
+
+const options: monaco.editor.IDiffEditorConstructionOptions = {
+    originalEditable: true,
+    diffCodeLens: true,
+    dragAndDrop: true,
+    tabCompletion: 'on',
+    renderSideBySide: true,
+};
+
+const FileUploader = ({
+                          onFileLoad,
+                          id,
+                          buttonText,
+                      }: {
+    onFileLoad: Function;
+    id: string;
+    buttonText: string;
+}) => {
+    return (
+        <>
+            <input
+                type="file"
+                accept="*/*"
+                id={id}
+                hidden
+                onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
+                    if (e?.target?.files && e.target.files.length) {
+                        return onFileLoad(e.target.files[0]);
+                    }
+                    return null;
+                }}
+            />
+            <label htmlFor={id}>
+                <Button variant="contained" color="primary" component="span">
+                    {buttonText}
+                </Button>
+            </label>
+        </>
+    );
+};
+
+function readFileAndSetText(
+    file: File | undefined,
+    setText: (value: ((prevState: string) => string) | string) => void,
+    setLanguage: (value: ((prevState: string) => string) | string) => void,
+    allowedLanguages: MonacoLanguages[],
+) {
+    if (file) {
+        const reader = new FileReader();
+        reader.onload = async e => {
+            // @ts-ignore
+            setText(e.target.result);
+        };
+        reader.readAsText(file);
+        let newLanguage = 'plaintext';
+        const extension = `.${file.name.split('.').pop()}`;
+        if (allowedLanguages?.length) {
+            for (let i = 0; i < allowedLanguages.length; i++) {
+                if (allowedLanguages[i].extensions.includes(extension as string)) {
+                    newLanguage = allowedLanguages[i].name;
+                    break;
+                }
+            }
+        }
+        setLanguage(newLanguage);
+    }
+}
+
+export type MonacoLanguages = { name: string; extensions: string[] };
+
+function Diff() {
+    const styles = useStyles();
+    const [originalFile, setOriginalFile] = useState<File>();
+    const [modifiedFile, setModifiedFile] = useState<File>();
+
+    const [originalText, setOriginalText] = useState(
+        `Backstage toolbox\n\ncompare text`,
```

Comment:
Don't set default texts, instead provide user a sample button like other tools to prevent unnecessary clearing of inputs